pkg/providers/vsphere/defaults.go

```diff
@@ -101,12 +101,27 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	return nil
 }
 
-func (d *Defaulter) setCloneModeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
+func max(a, b int) int {
```

Comment:
Why not just use the built-in `Math.Max()` function?

Reply:
[Math.Max ](https://pkg.go.dev/math#Max)operates on float64s only. The values being compared here are ints to it's would require a bunch of type conversions. This just makes things easier.

Reply:
Yea but you can pass ints too. It shouldn't change the result right?

Reply:
Passing ints doesn't work: https://go.dev/play/p/9xwiJu-dDGf
So I would have to convert the inputs to float64 and the output back to int, which doesn't look very clean. Here's what that would look like: https://go.dev/play/p/ylG5-_FhjFd

Reply:
I was confused with github.com/pkg/math, this is fine then